research/sentiment_analysis/data/imdb.py

```diff
@@ -0,0 +1,23 @@
+import tensorflow as tf, numpy as np
+from data.util import to_dataset, pad_sentence, START_CHAR, OOV_CHAR
+
+NUM_CLASS = 2
+_x_train, _y_train, _x_test, _y_test = None, None, None, None
```

Comment:
@qlzh727 The dataset for this model is not too big in size and creating in-memory dataset is not very heavy loaded job. I think splitting this dataset might be an overkill. Also, this is because I am not familiar with how tensorflow deals with dataset, wouldn't the tensorflow hold the whole bench if not the whole dataset in the memory? Holding the dataset here in memory is not a big issue, and swapping in/out disk might be over kill.

Reply:
The issue (as we only recently discovered) is that if you do in-memory tensorflow is forced to dump the entire dataset contents into the graph.